pkgs/build-support/flutter/default.nix

```diff
@@ -29,131 +17,54 @@ let
-
-    outputs = [ "out" "debug" ];
-
-    nativeBuildInputs = [
-      makeWrapper
-      deps
-      flutter
-      jq
-    ] ++ nativeBuildInputs;
-
-    preUnpack = ''
-      ${lib.optionalString (!autoDepsList) ''
-        if ! { [ '${lib.boolToString (depsListFile != null)}' = 'true' ] ${lib.optionalString (depsListFile != null) "&& cmp -s <(jq -Sc . '${depsListFile}') <(jq -Sc . '${finalAttrs.passthru.depsListFile}')"}; }; then
-          echo 1>&2 -e '\nThe dependency list file was either not given or differs from the expected result.' \
-                      '\nPlease choose one of the following solutions:' \
-                      '\n - Duplicate the following file and pass it to the depsListFile argument.' \
-                      '\n   ${finalAttrs.passthru.depsListFile}' \
-                      '\n - Set autoDepsList to true (not supported by Hydra or permitted in Nixpkgs)'.
-          exit 1
-        fi
-      ''}
-
-      ${preUnpack}
-    '';
-
-    configurePhase = ''
-      runHook preConfigure
-
-      ${flutterSetupScript}
-
-      runHook postConfigure
-    '';
-
-    buildPhase = ''
-      runHook preBuild
-
-      mkdir -p build/flutter_assets/fonts
-
-      doPubGet flutter pub get --offline -v
-      flutter build linux -v --release --split-debug-info="$debug" ${builtins.concatStringsSep " " (map (flag: "\"${flag}\"") finalAttrs.flutterBuildFlags)}
-
-      runHook postBuild
-    '';
-
-    installPhase = ''
-      runHook preInstall
+  dontDartBuild = true;
+  buildPhase = ''
+    runHook preBuild
 
-      built=build/linux/*/release/bundle
+    mkdir -p build/flutter_assets/fonts
 
-      mkdir -p $out/bin
-      mv $built $out/app
+    doPubGet flutter pub get --offline -v
+    flutter build linux -v --release --split-debug-info="$debug" ${builtins.concatStringsSep " " (map (flag: "\"${flag}\"") flutterBuildFlags)}
 
-      for f in $(find $out/app -iname "*.desktop" -type f); do
-        install -D $f $out/share/applications/$(basename $f)
-      done
+    runHook postBuild
+  '';
 
-      for f in $(find $out/app -maxdepth 1 -type f); do
-        ln -s $f $out/bin/$(basename $f)
-      done
+  dontDartInstall = true;
+  installPhase = ''
+    runHook preInstall
 
-      # make *.so executable
-      find $out/app -iname "*.so" -type f -exec chmod +x {} +
+    built=build/linux/*/release/bundle
 
-      # remove stuff like /build/source/packages/ubuntu_desktop_installer/linux/flutter/ephemeral
-      for f in $(find $out/app -executable -type f); do
-        if patchelf --print-rpath "$f" | grep /build; then # this ignores static libs (e,g. libapp.so) also
-          echo "strip RPath of $f"
-          newrp=$(patchelf --print-rpath $f | sed -r "s|/build.*ephemeral:||g" | sed -r "s|/build.*profile:||g")
-          patchelf --set-rpath "$newrp" "$f"
-        fi
-      done
+    mkdir -p $out/bin
+    mv $built $out/app
 
-      runHook postInstall
-    '';
+    for f in $(find $out/app -iname "*.desktop" -type f); do
```

Comment:
I think the desktop file search is alright, though I'm not sure why it's there to begin with.

As for the main binary search, determining the name of the executable in advance is non-trivial. We could use the `depsListFile` to find the `root` package, but this would involve a linear search and be quite inefficient. We could parse the `pubspec.yaml`, but this would require a YAML parser to be added to the build closure. I don't think it's very likely for this `find` invocation to go wrong, as I've never seen a case where other files are added to the `app` directory.

The third `find` cannot practically be done in another way, as the names of library files added by plugins are quite arbitrary.

@mkg20001 I have some questions:
- Why search for desktop files here? Is there a tool in the Flutter ecosystem that puts them in this location? If not, we can probably remove this in favor of `copyDesktopItems`.
- Why do the library files need to be executable?

Anyway, none of this is new in this PR, which is mainly a refactor. I think we can merge this PR first, and change the behavior in question later.

Reply:
ah totally missed this. most of this is hackery from my first packaging attempts. try to remove some and see if things break.

library files need to be executable, but some weren't. maybe that is fixed now.

as for the ln that's mostly duct tape.

desktop files: i think was firmware-updater that puts them there